sway/input/keyboard.c

```diff
@@ -476,13 +476,47 @@ struct sway_keyboard *sway_keyboard_create(struct sway_seat *seat,
 	return keyboard;
 }
 
-struct xkb_keymap *sway_keyboard_compile_keymap(struct input_config *ic) {
+static void handle_xkb_context_log(struct xkb_context *context,
+		enum xkb_log_level level, const char *format, va_list args) {
+	va_list args_copy;
+	va_copy(args_copy, args);
+	size_t length = vsnprintf(NULL, 0, format, args_copy) + 1;
+	va_end(args_copy);
+
+	char *error = malloc(length);
+	if (!error) {
+		sway_log(SWAY_ERROR, "Failed to allocate libxkbcommon log message");
+		return;
+	}
+
+	va_copy(args_copy, args);
+	vsnprintf(error, length, format, args_copy);
+	va_end(args_copy);
+
+	if (error[length - 2] == '\n') {
+		error[length - 2] = '\0';
+	}
+
+	sway_log(SWAY_ERROR, "(xkbcommon) %s", error);
```

Comment:
Maybe we can use `level` to choose the sway log level?

Reply:
Done